oauth2_http/java/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentials.java

```diff
@@ -210,10 +219,25 @@ public AccessToken refreshAccessToken() throws IOException {
       this.refreshToken = refreshToken;
     }
 
-    return AccessToken.newBuilder()
-        .setExpirationTime(expiresAtMilliseconds)
-        .setTokenValue(accessToken)
-        .build();
+    AccessToken newAccessToken =
+        AccessToken.newBuilder()
+            .setExpirationTime(expiresAtMilliseconds)
+            .setTokenValue(accessToken)
+            .build();
+
+    refreshTrustBoundary(newAccessToken, transportFactory);
+    return newAccessToken;
+  }
+
+  @Override
+  public String getTrustBoundaryUrl() throws IOException {
+    Matcher matcher = WORKFORCE_PATTERN.matcher(getAudience());
+    if (!matcher.matches()) {
+      throw new IOException(
+          "The provided audience is not in the correct format for a workforce pool.");
```

Comment:
nit: Maybe IllegalStateException would be better here? I don't think there is an IO going on here when `getAudience()` is called.